### PR TITLE
Update with all documentation notes by Lucan

### PR DESCRIPTION
**Pull request by Luashine**
Transferred all explanations from the well known document:
"Blz native Documentation (new natives and more)"
https://docs.google.com/document/d/1AjAScGNEgT8puu5rn0rdeTr2TWxSz3BIlWdcfDCUoLA/
by Lycan#7531@Discord, the permission was granted

Otherwise minor changes and adaptations where descriptions existed

---------

I have not run the the build, please (do it for me) verify that everything's alright. Please review the changes and feedback. Finishing this the only other centralized place of documentation will be moyack's [jassdoc](https://wc3modding.info/pages/jass-documentation-database/) (and of course small pieces on forums of thehelper/thehive)

I'd also like to open a discussion about style, new keywords to reduce the copy-paste burden etc... in Issues or in the Hive's topic? Discussing over at the Hive won't be a bad idea, we'd keep bumping the thread with every response :)

### CODE REVIEW

File: ability.j [hunk 1]
```diff
@@ -52,82 +68,126 @@ Supports Unit/Item/Ability/Tech Codes
...
+
 /**
+Change(Set) the ability Y coordinate (Cartesian System) of the activated ability icon in the default 4x3 grid.
+
 @patch 1.29
 */
 native BlzSetAbilityActivatedPosY                  takes integer abilCode, integer y returns nothing
+
 /**
+Requires an ability ID and the ability level and returns the ability’s (at the level passed) mana cost. *Since 1.31: use Level 0 to read manacosts from Level 1.*
```
Comment (lep) [inline ability.j:182]: Use `@note`

File: ability.j [hunk 2]
```diff
@@ -52,82 +68,126 @@ Supports Unit/Item/Ability/Tech Codes
...
+
 @patch 1.29
 */
 native BlzGetAbilityManaCost                       takes integer abilId, integer level returns integer
+
 /**
+Requires an ability ID and the ability level and returns the ability’s (at the level passed) cooldown. *Since 1.31: use Level 0 to read cooldown from Level 1.*
```
Comment (lep) [inline ability.j:189]: Use `@note` for the patch stuff.

File: ability.j (hunk 1 above)
Comment (lep) [inline ability.j:182]: Use `@note` for the patch stuff.

File: effects.j [hunk 3]
```diff
@@ -307,54 +334,93 @@ native BlzSetSpecialEffectY                        takes effect whichEffect, rea
...
+
 @async
 @patch 1.29
 */
 native BlzGetLocalSpecialEffectZ                   takes effect whichEffect returns real
 
 /**
+Clears all of the subanimations (tags) of a specified special effect. (It does not affect normal animations)
+
+**Example usage of subanimations:**
```
Comment (lep) [inline effects.j:370]: Use markdown code-blocks here (just indent by four spaces)

File: effects.j [hunk 4]
```diff
@@ -307,54 +334,93 @@ native BlzSetSpecialEffectY                        takes effect whichEffect, rea
...
+
+1. *Overrides the currently playing animation/subanimation.*
+2. *TimeScale is a real, meaning that it can be both negative and positive numbers with decimals, there are examples in which you can use negative numbers mid animation to make it go backwards, however in this case it starts at 0 meaning that it can’t be negative.*
+
 @patch 1.30
 */
 native BlzPlaySpecialEffectWithTimeScale           takes effect whichEffect, animtype whichAnim, real timeScale returns nothing
 
 /**
+Receives an animtype (type, extends handle), and returns the string representation of the name of the animation.
```
Comment (lep) [inline effects.j:422]: You can put any jass code (the `animtype` and `extends handle`) in backticks `` ` `` so we get some nice linkage on the web view.

File: misc.j [hunk 5]
```diff
@@ -121,21 +121,29 @@ native AutomationTestEnd                        takes nothing returns nothing
 native AutomationTestingFinished                takes nothing returns nothing
 
 /**
+According to MindWorX (Blizzard Developer): //They do nothing of interest //Just a compatibility thing
```
Comment (lep) [inline misc.j:124]: If you have a linkable source you can include the link but i would just rewrite without the slashes and "accoring to", etc.

File: ui.j [hunk 6]
```diff
@@ -21,36 +30,67 @@ native BlzIsSelectionCircleEnabled                 takes nothing returns boolean
...
+ORIGIN_FRAME_HERO_BUTTON [0 to 6] The clickable hero icons at the left of the screen
+ORIGIN_FRAME_HERO_HP_BAR [0 to 6] ^^
+ORIGIN_FRAME_HERO_MANA_BAR [0 to 6] ^^
+ORIGIN_FRAME_HERO_BUTTON_INDICATOR [0 to 6] ^^
+ORIGIN_FRAME_ITEM_BUTTON [0 to 5] the Item Buttons
+ORIGIN_FRAME_MINIMAP_BUTTON The buttons altering the minimap
+ORIGIN_FRAME_SYSTEM_BUTTON [0 to 3] {Menu, Allies, Log, Quest}
+
+Here is a basic example that creates a custom timerdialog window:
+set GameUI = BlzGetOriginFrame(ORIGIN_FRAME_GAME_UI, 0)
```
Comment (lep) [inline ui.j:61]: Put an empty line before the example code and indent it.

File: ui.j [hunk 7]
```diff
@@ -21,36 +30,67 @@ native BlzIsSelectionCircleEnabled                 takes nothing returns boolean
...
+set UIMain = BlzCreateFrame("TimerDialog", GameUI, 0, 0)
+call BlzFrameSetPoint(UIMain, FRAMEPOINT_CENTER, GameUI, FRAMEPOINT_CENTER, 0.25, 0.055)
+call BlzFrameSetSize(UIMain, 0.3, 0.7)
+
+*Take a look at the .fdf files in the game’s CASC or point 1.3 (refer to this document’s table of contents for reference) it should give you some ideas.*
+
+Example code:
+BlzHideOriginFrames(true)
```
Comment (lep) [inline ui.j:69]: Put an empty line before the example code and indent it.

File: ui.j [hunk 8]
```diff
@@ -137,35 +181,60 @@ Useful to move frames with the next SetPoint.
 native BlzFrameClearAllPoints                      takes framehandle frame returns nothing
 
 /**
+TODO
```
Comment (lep) [inline ui.j:184]: Remove TODO. If it's a bigger Task maybe open an issue.

File: ui.j [hunk 9]
```diff
@@ -137,35 +181,60 @@ Useful to move frames with the next SetPoint.
...
+
+Example:
```
Comment (lep) [inline ui.j:186]: Put an empty line before the example code and indent it.

File: ui.j [hunk 10]
```diff
@@ -137,35 +181,60 @@ Useful to move frames with the next SetPoint.
...
 native BlzFrameIsVisible                           takes framehandle frame returns boolean
 
 
 /**
+Requires a string for the frame name that you want to retrieve (get), and an integer (which in most cases should be 0) that specifies the index of the frame that you want to get (for example for inventory slots you have 6, from 0-5).
+
 Read from the internal Frame-Storage.
 The first time a Frame enters the map's script it takes a handleId.
 
+Example:
```
Comment (lep) [inline ui.j:221]: For one-line examples you don't need an entire block just wrap it in backticks.

File: ui.j [hunk 11]
```diff
@@ -282,32 +353,41 @@ Affects child-Frames, when they don't have an own Scale.
...
+Sets the current Value, only for FrameType that use this feature:
 POPUPMENU, SLIDER, SIMPLESTATUSBAR, STATUSBAR
+
 @patch 1.31
 */
 native BlzFrameSetValue                            takes framehandle frame, real value returns nothing
 
 /**
-Gets the current local Value.
+Gets the current local Value. Result can differ per player
```
Comment (lep) [inline ui.j:389]: Technically the async tag already states that values can differ per player but if you think it's valuable to highlight it i don't mind.

File: unit-based-event-api.j [hunk 12]
```diff
@@ -33,48 +33,90 @@ constant native GetEventDamageSource takes nothing returns unit
 
 
 /**
+Set the damage amount of a damage event.
+
+In 1.31 PTR there’s currently 3 new damage events:
+1. EVENT_UNIT_DAMAGED - old classic event for a specific unit;
```
Comment (lep) [inline unit-based-event-api.j:39]: Feel free to wrap the event names in backticks.

File: unit-based-event-api.j [hunk 13]
```diff
@@ -33,48 +33,90 @@ constant native GetEventDamageSource takes nothing returns unit
...
+2. EVENT_PLAYER_UNIT_DAMAGED - Same as 1, but for all units of a specific player on the map;
+   // This seems to work fine anyway lol:
+   call TriggerRegisterAnyUnitEventBJ(gg_trg_a, EVENT_PLAYER_UNIT_DAMAGING)
```
Comment (lep) [inline unit-based-event-api.j:42]: Code in backticks.

File: unit-based-event-api.j [hunk 14]
```diff
@@ -33,48 +33,90 @@ constant native GetEventDamageSource takes nothing returns unit
...
+
+1 and 2 - modify the damage after any reduction.
+3 and 4 - changes damage before reduction. Amount you set will be reduced later according to target’s resistance, armor etc.
+
+If set to <=0 during 3 or 4, then 1 or 2 will never fire.
+Misses don’t trigger any damage events.
+Set to 0.00 to completely block the damage.
+Set to negative value to heal the target instead of damaging
+
+@note Tip: calling GetEventDamage() after you set it with this function will return the value you set.
```
Comment (lep) [inline unit-based-event-api.j:54]: Code in backticks.

File: unit-based-event-api.j [hunk 15]
```diff
@@ -33,48 +33,90 @@ constant native GetEventDamageSource takes nothing returns unit
...
+
 @event EVENT_UNIT_DAMAGED
 @patch 1.29
 */
 native BlzSetEventDamage                           takes real damage returns nothing
 
 /**
+The target unit of the damage event.
+If damage is AoE, your trigger will run separately for each target without known issues.
+This returns the same result as GetTriggerUnit
```
Comment (lep) [inline unit-based-event-api.j:65]: Code in backticks.

File: unit-based-event-api.j [hunk 16]
```diff
@@ -33,48 +33,90 @@ constant native GetEventDamageSource takes nothing returns unit
...
+
 @event EVENT_UNIT_DAMAGED
 @patch 1.31
 */
 native BlzGetEventDamageTarget 	                   takes nothing returns unit
 
 /**
+Returns attacktype of the damage being taken.
+Spell-damage is ATTACK_TYPE_NORMAL
```
Comment (lep) [inline unit-based-event-api.j:74]: Code in backticks.

File: unit-based-event-api.j [hunk 17]
```diff
@@ -33,48 +33,90 @@ constant native GetEventDamageSource takes nothing returns unit
...
+Spell-damage is ATTACK_TYPE_NORMAL 
+
 @event EVENT_UNIT_DAMAGED
 @patch 1.31
 */
 native BlzGetEventAttackType  	                   takes nothing returns attacktype
 
 /**
+Returns damagetype of the damage being taken.
+Regular attack is DAMAGE_TYPE_NORMAL.
```
Comment (lep) [inline unit-based-event-api.j:83]: Code in backticks.

File: unit.j [hunk 18]
```diff
@@ -1036,6 +1139,21 @@ native BlzGetUnitStringField                       takes unit whichUnit, unitstr
 native BlzSetUnitBooleanField                      takes unit whichUnit, unitbooleanfield whichField, boolean value returns boolean
 
 /**
+Problems:
+`BlzSetUnitIntegerField(unit, UNIT_IF_HIT_POINTS_REGENERATION_TYPE`
```
Comment (lep) [inline unit.j:1143]: Problems? Also code misses a closing `)`

File: unit.j [hunk 19]
```diff
@@ -1102,6 +1225,8 @@ native BlzSetUnitWeaponRealField                   takes unit whichUnit, unitwea
 native BlzSetUnitWeaponStringField                 takes unit whichUnit, unitweaponstringfield whichField, integer index, string value returns boolean
 
 /**
+This does not update `IsUnitPaused()` and keeps the command card visible. Otherwise identical to `PauseUnit()`.
```
Comment (lep) [inline unit.j:1228]: Personally i would remove the `()` from natives but in the end i don't mind.

**Comment by lep** [comment]
Big thanks. I added a bunch of comments. But that's mostly style and/or technicalities. The important part is the content which you've provided a lot of.
Re discussion: i don't care if we do it here or on hive.

File: unit.j (hunk 18 above)
Comment (Luashine) [reply unit.j:1143]: Lucan used "Problems:" a few times, looks like the bugs/peculiarities he discovered and hadnt had time to explore in depth. I'll have rephrased it in the upcoming commit, but generally these places should be regarded as "Work in progress"

**Comment by Luashine** [comment]
Thanks this was exactly what I wanted. My excuse is: 1. not knowing exactly when to use what :) 2. to develop a feeling for what a unified formatting style should be later on, since I just wanted to put all of his notes in one place for now.

I dont think to have missed anything, github collapsed 10 of your comments and I had to go over everything once again when I noticed...

File: ui.j (hunk 11 above)
Comment (Luashine) [reply ui.j]: I didn't notice that the async tag already has a very visible warning on the doc page. Done

File: ui.j [hunk 20]
```diff
@@ -45,29 +45,41 @@ native BlzIsTargetIndicatorEnabled                 takes nothing returns boolean
...
 
 *Take a look at the .fdf files in the game’s CASC or point 1.3 (refer to this document’s table of contents for reference) it should give you some ideas.*
 
-Example code:
-BlzHideOriginFrames(true)
-BlzFrameSetAllPoints(BlzGetOriginFrame(E, 0)ORIGIN_FRAME_WORLD_FRAM, BlzGetOriginFrame(ORIGIN_FRAME_GAME_UI, 0))
+Example:
+
+    BlzHideOriginFrames(true)
+    BlzFrameSetAllPoints(BlzGetOriginFrame(E, 0)ORIGIN_FRAME_WORLD_FRAM, BlzGetOriginFrame(ORIGIN_FRAME_GAME_UI, 0))
```
Comment (lep) [inline ui.j:82]: This code seems wrong.

**Review by lep — COMMENTED** [review]
One small change requested. Just let me know when you want it to be merged. Once it's merged i/we can fix style mistakes as we find them.

File: ui.j (hunk 20 above)
Comment (Luashine) [reply ui.j:82]: Hm it appears exactly like that in his doc. Seems that `ORIGIN_FRAME_WORLD_FRAM` should've been removed, then the code looks like to be correct. I'm gonna ask him or remove that example entirely, I'll push a commit later. Good catch

Comment (Luashine) [reply ui.j:82]: Lycan doesn't know anymore, removed that piece entirely.

**Comment by Luashine** [comment]
Ready to merge from me :ok_hand:

**Pull request merged**